File: app/models/helper/TmdbMovie.java
package models.helper;

import java.util.List;

import javax.persistence.Embedded;

import org.joda.time.format.DateTimeFormat;
import org.joda.time.format.DateTimeFormatter;

import play.Logger;

import com.google.gson.annotations.SerializedName;

public class TmdbMovie{
	
	public long id;
	
	@SerializedName("imdb_id")
	public String imdbId;
	public String name;
	public String url;
	public int popularity;
	public boolean translated;
	public boolean adult;
	public String language;
	@SerializedName("original_name")
	public String originalName;
	@SerializedName("alternative_name")
	public String alternativeName;
	@SerializedName("movie_type")
	public String movieType;
	
	public int votes;
	public String rating;
	public String certification;
	public String overview;
	public String released;

	public int version;
	@SerializedName("last_modified_at")
	public String lastModifiedAt;
	
	@Embedded
	public List<TmdbPoster> posters;
	
	@Embedded
	public List<TmdbBackDrop> backdrops;
    
    public Integer getYear() {
    	if(released == null){
    		return null;
    	}
		DateTimeFormatter fmt = DateTimeFormat.forPattern("yyyy-MM-dd");
		try{
			return fmt.parseDateTime(released).getYear();
		}catch(IllegalArgumentException ex){
			return null;
		}
	}
    
	public int getPercentRating(){
		if(rating == null){
			return -1;
		}
		try{
			return (int)(Double.parseDouble(rating) * 10);
		} catch (NumberFormatException e) {
			Logger.warn(e.getMessage());
		}
		return -1;
	}
	
	public int getDecimalRating(){
		return getPercentRating() / 10;
	}
	
	@Override
	public String toString() {
		return id + " " + name;
	}
}

File: app/models/helper/TmdbMovie.scala
package models.helper

import java.time.LocalDate
import java.time.format.DateTimeFormatter

import scala.util.Try
import TmdbMovie._

object TmdbMovie{
  private final val fmt = DateTimeFormatter.ofPattern("yyyy-MM-dd")
}

case class TmdbMovie(
    
	id:Long,

  imdb_id:String,
	
	name:String,
	url:String,
	popularity:Int,
	translated:Boolean,
	adult:Boolean,
	language:String,
  original_name:String,
  alternative_name:String,
  movie_type:String,
	
	votes:Int,
	rating:Option[String],
	certification:String,
	overview:String,
	released: Option[String],

	version:Int,
  last_modified_at:String,
	
	posters:List[TmdbPoster],
	
	backdrops:List[TmdbBackDrop]
) {

  val imdbId = imdb_id
  val originalName = original_name
  val alternativeName = alternative_name
  val movieType = movie_type
  val lastModifiedAt = last_modified_at

  def getFirstPoster = posters.headOption.map(_.image.url)

  def getYear = {
    released.flatMap { rel =>

      Try(LocalDate.parse(rel, fmt).getYear).toOption
    }
  }
    
	def getPercentRating = {
    rating.flatMap(rat => Try(rat.toDouble).toOption)
	}
	
	def getDecimalRating = getPercentRating.map(_ / 10)

}
